service/storage_proxy.cc

```diff
@@ -1348,16 +1348,13 @@ endpoints_to_replica_ids(const locator::token_metadata& tm, const inet_address_v
 }
 
 query::max_result_size storage_proxy::get_max_result_size(const query::partition_slice& slice) const {
-    // Unpaged and reverse queries.
-    // FIXME: some reversed queries are no longer unlimited.
-    // To filter these out we need to know if it's a single partition query.
-    // Take the partition key range as a parameter? This would require changing many call sites
-    // where the partition key range is not immediately available.
-    // For now we ignore the issue, return the 'wrong' limit for some reversed queries, and wait until all reversed queries are fixed.
+    auto page_size = _features.cluster_supports_reduced_revenced_queries_limit() ? query::result_memory_limiter::maximum_result_size : 0;
     if (!slice.options.contains<query::partition_slice::option::allow_short_read>() || slice.options.contains<query::partition_slice::option::reversed>()) {
-        return _db.local().get_unlimited_query_max_result_size();
+        auto max_size = _db.local().get_unlimited_query_max_result_size();
+        max_size.page_size = page_size;
+        return max_size;
     } else {
-        return query::max_result_size(query::result_memory_limiter::maximum_result_size);
+        return query::max_result_size(query::result_memory_limiter::maximum_result_size, query::result_memory_limiter::maximum_result_size, page_size);
     }
```

Comment:
As we discussed in the earlier PR, if the feature is set we should unconditionally return the scheduling-limit appropriate limit + page size:
```
if (_features.cluster_supports_reduced_revenced_queries_limit()) {
    auto max_size =  _db.local().get_unlimited_query_max_result_size();
    max_size.page_size = query::result_memory_limiter::maximum_result_size;
    return max_size;
}
// old-code as-is

Reply:
Isn't current implementation doing exactly the same?

Reply:
The current implementation doesn't fill the limit fields for normal queries and I think we should move away from the practice of selecting which fields to fill based on query-type.
Also I think what I proposed is nicer because it results in much more readable code and has a much clearer separation of the behavior pre and post feature. It also allows us to simply remove the if and drop the old code in 2 years.

Reply:
But the current implementation does fill the limit fields. That's what the single argument constructor does.

The argument of dropping the rest of the code in the future makes sense. Otherwise, it really does exactly the same thing.

Reply:
It does but with all the same values:
```c++
return query::max_result_size(query::result_memory_limiter::maximum_result_size, query::result_memory_limiter::maximum_result_size, page_size);
```
I wanted to use the real-deal returned by `_db.local().get_unlimited_query_max_result_size();` for the limits in all cases.